src/linux/Packaging.Linux/install-from-source.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
```

Comment:
Do we want the script print a message explaining what it does, and prompt:

```shell
This script will download, compile, and install Git Credential Manager to:

  ~/usr/local/bin

Git Credential Manager is licensed under the MIT License: https://aka.ms/gcm/license

Do you want to continue? [Y/n]
```

Might hinder automation; so adding a `--yes` option to skip would be required here too.

Could also do all our pre-checks here for missing/required dependencies:

```shell
This script will download, compile, and install Git Credential Manager to:

  ~/usr/local/bin

Git Credential Manager is licensed under the MIT License: https://aka.ms/gcm/license

ERROR: There are missing dependencies required to continue:
 * git
 * curl
Please install these and try again.
```

Reply:
I'm a little confused - we do attempt to install `git` and `curl` for all distributions when we call the `install_shared_packages()` method. Should we not be doing this and instead warn users as you indicated?

Reply:
Ah right, you're correct! I had missed that when I wrote this comment - sorry! 🤦‍♂️ 

> Should we not be doing this and instead warn users as you indicated?

I wonder if _always running_ `<pkgmgr> <installcmd> curl git` is a good idea for those who are using custom-built Git's, or other such setups?
We are already doing a check for `dotnet` and the appropriate SDKs/versions before attempting to install, why not also `git` and `curl`?

Even just a simple `which -s git` and `which -s curl` could be enough not to stomp over someones setup? Personally if a script stomped on my setup I'd not be happy 😁

Reply:
I added a version of this in the latest iteration. Note that I did not use the `-s` flag because that resulted in the following during my testing:

```shell
ldennington@ubuntu:~$ which -s git
Illegal option -s
Usage: /usr/bin/which [-a] args
```